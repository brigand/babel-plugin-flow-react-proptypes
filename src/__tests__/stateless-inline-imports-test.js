const babel = require('babel-core');
const content = `
// @flow

import React from 'react';
import type { T } from '../types';

const C = (props: T) => {
  <div>{props.name}</div>
};
`;

it('stateless-inline-imports', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
