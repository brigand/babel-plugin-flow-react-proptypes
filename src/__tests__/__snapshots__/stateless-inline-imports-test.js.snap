// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateless-inline-imports 1`] = `
"\\"use strict\\";

var _react = _interopRequireDefault(require(\\"react\\"));

var _types = require(\\"../types\\");

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

var C = function C(props) {
  /*#__PURE__*/
  _react[\\"default\\"].createElement(\\"div\\", null, props.name);
};

C.propTypes = _types.bpfrpt_proptype_T === _propTypes[\\"default\\"].any ? {} : _types.bpfrpt_proptype_T;"
`;
